packages/core/src/shared/utilities/zipStream.ts

```diff
@@ -104,31 +103,44 @@ export class ZipStream {
             })
         }
     }
-
-    public writeString(data: string, path: string) {
-        return this._zipWriter.add(path, new TextReader(data))
+    /**
+     * Writes data to the specified path.
+     * @param data
+     * @param path
+     * @param returnPromise optional parameter specifying if caller wants a promise.
+     * @returns promise to that resolves when data is written.
+     */
+    public writeString(data: string, path: string, returnPromise: true): Promise<EntryMetaData>
+    public writeString(data: string, path: string, returnPromise?: false): void
+    public writeString(data: string, path: string, returnPromise?: boolean): void | Promise<EntryMetaData> {
+        const promise = this._zipWriter.add(path, new TextReader(data))
```

Comment:
Whats the use case for the promise/non promise response vs just always returning a promise and then callers can either do await or void

Reply:
My understanding of the `ZipStream` code is that it will wait until all files are done streaming in the `finalize` step. Therefore the most common use case is to avoid awaiting the promises individually so that we aren't waiting on IO operations until all files are added. Its only the tests where we await these promises.